plenum/server/batch_handlers/node_reg_handler.py

```diff
@@ -106,6 +114,12 @@ def commit_batch(self, three_pc_batch: ThreePcBatch, prev_handler_result=None):
         for view_no in view_nos[:prev_committed_index]:
             self.node_reg_at_beginning_of_view.pop(view_no, None)
 
+        # trigger view change if nodes count changed
+        # TODO: create a new message to pass Suspicious events and make ViewChangeTriggerService the only place for
+        # view change triggering
+        if self.internal_bus and len(prev_committed) != len(self.committed_node_reg):
```

Comment:
Maybe some future optimization. For case of reordering will the message with current view_no be sent?

Reply:
Yes, in case of re-ordering it will be the original_view_no + 1.
So, it will not take effect for the nodes who already moved to this view, but may help out of date nodes to start view change

Reply:
i agree.